File: mobile/modules/vibe-chat-native/ios/ChatNativeProfileModule.swift
import ExpoModulesCore

public class ChatNativeProfileModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ChatNativeProfile")

    Function("isSupported") {
      true
    }

    Function("supportsNativeProfile") {
      true
    }

    View(ChatProfileMainView.self) {
      Prop("profileOnly") { (view: ChatProfileMainView, value: Bool?) in
        view.setProfileOnly(value ?? true)
      }

      Prop("surfaceId") { (view: ChatProfileMainView, value: String) in
        view.surfaceId = value
      }

      Prop("rows") { (view: ChatProfileMainView, rows: [[String: Any]]) in
        view.setRows(rows)
      }

      Prop("engineSurfaceId") { (view: ChatProfileMainView, value: String?) in
        view.setEngineSurfaceId(value ?? "")
      }

      Prop("chatId") { (view: ChatProfileMainView, value: String?) in
        view.setEngineChatId(value ?? "")
      }

      Prop("myUserId") { (view: ChatProfileMainView, value: String?) in
        view.setEngineMyUserId(value ?? "")
      }

      Prop("peerUserId") { (view: ChatProfileMainView, value: String?) in
        view.setEnginePeerUserId(value ?? "")
      }

      Prop("statusAuthorityEnabled") { (view: ChatProfileMainView, enabled: Bool) in
        view.setStatusAuthorityEnabled(enabled)
      }

      Prop("appearance") { (view: ChatProfileMainView, appearance: [String: Any]) in
        view.setAppearance(appearance)
      }

      Prop("headerTitle") { (view: ChatProfileMainView, value: String?) in
        view.setHeaderTitle(value ?? "")
      }

      Prop("headerSubtitle") { (view: ChatProfileMainView, value: String?) in
        view.setHeaderSubtitle(value ?? "")
      }

      Prop("profileName") { (view: ChatProfileMainView, value: String?) in
        view.setProfileName(value ?? "")
      }

      Prop("profileHandle") { (view: ChatProfileMainView, value: String?) in
        view.setProfileHandle(value ?? "")
      }

      Prop("profileBio") { (view: ChatProfileMainView, value: String?) in
        view.setProfileBio(value ?? "")
      }

      Prop("avatarUri") { (view: ChatProfileMainView, value: String?) in
        view.setAvatarUri(value)
      }

      Prop("isOnline") { (view: ChatProfileMainView, value: Bool?) in
        view.setIsOnline(value ?? false)
      }

      Prop("isChatMuted") { (view: ChatProfileMainView, value: Bool?) in
        view.setIsChatMuted(value ?? false)
      }

      Prop("isGroupOrChannel") { (view: ChatProfileMainView, value: Bool?) in
        view.setIsGroupOrChannel(value ?? false)
      }

      Prop("groupMembers") { (view: ChatProfileMainView, value: [[String: Any]]?) in
        view.setGroupMembers(value ?? [])
      }

      Prop("groupMemberCount") { (view: ChatProfileMainView, value: Int?) in
        view.setGroupMemberCount(value)
      }

      Prop("agentConfig") { (view: ChatProfileMainView, value: [String: Any]?) in
        view.setAgentConfig(value)
      }

      Prop("page") { (view: ChatProfileMainView, value: String?) in
        guard
          let value,
          !value.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
        else { return }
        view.setPage(value, animated: true)
      }

      Events("onViewportChanged", "onNativeEvent")
    }
  }
}

File: mobile/modules/vibe-chat-native/android/src/main/java/expo/modules/vibechatnative/ChatNativeProfileModule.kt
package expo.modules.vibechatnative

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ChatNativeProfileModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ChatNativeProfile")

    Function("isSupported") {
      true
    }

    Function("supportsNativeProfile") {
      true
    }

    View(ChatProfileMainView::class) {
      Events("onViewportChanged", "onNativeEvent")

      Prop("profileOnly") { view: ChatProfileMainView, value: Boolean? ->
        view.setProfileOnly(value ?: true)
      }

      Prop("surfaceId") { view: ChatProfileMainView, value: String ->
        view.setSurfaceId(value)
      }

      Prop("rows") { view: ChatProfileMainView, rows: List<Map<String, Any?>> ->
        view.setRows(rows)
      }

      Prop("engineSurfaceId") { view: ChatProfileMainView, value: String? ->
        view.setEngineSurfaceId(value ?: "")
      }

      Prop("chatId") { view: ChatProfileMainView, value: String? ->
        view.setEngineChatId(value ?: "")
      }

      Prop("myUserId") { view: ChatProfileMainView, value: String? ->
        view.setEngineMyUserId(value ?: "")
      }

      Prop("peerUserId") { view: ChatProfileMainView, value: String? ->
        view.setEnginePeerUserId(value ?: "")
      }

      Prop("statusAuthorityEnabled") { view: ChatProfileMainView, enabled: Boolean ->
        view.setStatusAuthorityEnabled(enabled)
      }

      Prop("appearance") { view: ChatProfileMainView, appearance: Map<String, Any?> ->
        view.setAppearance(appearance)
      }

      Prop("headerTitle") { view: ChatProfileMainView, value: String? ->
        view.setHeaderTitle(value ?: "")
      }

      Prop("headerSubtitle") { view: ChatProfileMainView, value: String? ->
        view.setHeaderSubtitle(value ?: "")
      }

      Prop("profileName") { view: ChatProfileMainView, value: String? ->
        view.setProfileName(value ?: "")
      }

      Prop("profileHandle") { view: ChatProfileMainView, value: String? ->
        view.setProfileHandle(value ?: "")
      }

      Prop("profileBio") { view: ChatProfileMainView, value: String? ->
        view.setProfileBio(value ?: "")
      }

      Prop("avatarUri") { view: ChatProfileMainView, value: String? ->
        view.setAvatarUri(value)
      }

      Prop("isOnline") { view: ChatProfileMainView, value: Boolean? ->
        view.setIsOnline(value ?: false)
      }

      Prop("isChatMuted") { view: ChatProfileMainView, value: Boolean? ->
        view.setIsChatMuted(value ?: false)
      }

      Prop("isGroupOrChannel") { view: ChatProfileMainView, value: Boolean? ->
        view.setIsGroupOrChannel(value ?: false)
      }

      Prop("groupMembers") { view: ChatProfileMainView, value: List<Map<String, Any?>>? ->
        view.setGroupMembers(value ?: emptyList())
      }

      Prop("groupMemberCount") { view: ChatProfileMainView, value: Int? ->
        view.setGroupMemberCount(value)
      }

      Prop("agentConfig") { view: ChatProfileMainView, value: Map<String, Any?>? ->
        view.setAgentConfig(value)
      }

      Prop("page") { view: ChatProfileMainView, value: String? ->
        value?.let { view.setPage(it, true) }
      }
    }
  }
}
